lib/versioning/regex/index.ts

```diff
@@ -0,0 +1,151 @@
+const semver = require('semver');
+const { VersioningApi, RangeStrategy } = require('../common');
+const { logger } = require('../../logger');
+
+interface Version {
+  // major, minor, and patch act in the standard semver fashion, but without
+  // correctness requirements: if any one or two is omitted, they are treated
+  // as zero. At least one must be present
+  major: number;
+  minor: number;
+  patch: number;
+  // prereleases are treated in the standard semver manner, if present
+  prerelease: string;
+  // architectures, if present, are treated as a compatibility layer: we will
+  // never try to update to a version with a different architecture. Other than
+  // for juding compatibility (exact string match), the architecture is ignored
+  architecture: string;
+}
+
+// convenience method for passing a Version object into any semver.* method.
+function asSemver(version: Version): string {
+  let vstring = `${version.major}.${version.minor}.${version.patch}`;
+  if (typeof version.prerelease !== 'undefined') {
+    vstring += `-${version.prerelease}`;
+  }
+  return vstring;
+}
+
+export const api: VersioningApi = {
+  // config is expected to be overridden by a user-specified RegExp value
+  // sample values:
+  //
+  // * emulates the "semver" configuration:
+  //   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(-(?<prerelease>.*))?$')
+  // * emulates the "docker" configuration:
+  //   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(-(?<architecture>.*))?$')
+  // * matches the versoining scheme used by the Python images on DockerHub:
+  //   RegExp('^(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?<prerelease>[^.-]+)?(-(?<architecture>.*))?$');
+  //
+  // defaults to a simple semver scheme
+  config: null,
+
+  // TODO: the linter expects method shorthands for all these methods, but
```

Comment:
@ anyone-who-knows-javascript: Is there a way to follow this lint rule without breaking my usage of `this`?

@ maintainers: if not, can this rule be disabled?

Reply:
move `config` to module scope variable, then you dont have to use `this`. There can only be one instance of a versioning scheme.

Reply:
Ah perfect, that did the trick. Not sure why I thought I needed this in the first place -- thanks!